src/organizations/components/DeleteOrgOverlay.tsx

```diff
@@ -18,22 +18,80 @@ import {
   JustifyContent,
   Overlay,
 } from '@influxdata/clockface'
+import {track} from 'rudder-sdk-js'
 
 // Utils
 import {deleteAccount} from 'src/client/unityRoutes'
 import {notify} from 'src/shared/actions/notifications'
 import {accountSelfDeletionFailed} from 'src/shared/copy/notifications'
+import CancellationReasonsDropdown from './CancellationReasonsForm'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {event} from 'src/cloud/utils/reporting'
+
+// Selectors
+import {getQuartzMe} from 'src/me/selectors'
+import {DeleteOrgContext} from './DeleteOrgContext'
+import {getOrg} from 'src/organizations/selectors'
 
 const DeleteOrgOverlay: FC = () => {
   const history = useHistory()
   const [hasAgreedToTerms, setHasAgreedToTerms] = useState(false)
   const dispatch = useDispatch()
+  const [formErrors, setFormErrors] = useState({})
+  const {shortSuggestion, isShortSuggestionEnabled, suggestions} = useContext(
+    DeleteOrgContext
+  )
+  const quartzMe = useSelector(getQuartzMe)
+  const org = useSelector(getOrg)
 
   const handleClose = () => {
     history.goBack()
   }
 
+  const isFormValid = () => {
+    if (!isFlagEnabled('trackCancellations')) {
+      return true
+    }
+
+    const errors = {}
+    let hasErrors = false
+    if (isShortSuggestionEnabled && !shortSuggestion) {
+      errors['shortSuggestion'] = 'Please enter a suggestion'
+      hasErrors = true
+    }
+
+    if (!suggestions) {
+      errors['suggestions'] = 'Please enter suggestions'
+      hasErrors = true
+    }
+
+    setFormErrors(errors)
+
+    return !hasErrors
+  }
+
+  const sendDetailsToRudderstack = () => {
+    const payload = {
+      org: org.id,
+      tier: quartzMe?.accountType,
+      email: quartzMe?.email,
```

Comment:
getting the accountType here seems a bit redundant since the feature is only available for free customers right? If that's the case, we could also just get the email from the `user`, no need to add in another spot to get and set quartz data

Reply:
we do need this as we are going to add a similar thing for PAYG with the same payload